src/common/survival_util.cc

```diff
@@ -18,6 +18,109 @@
   https://github.com/avinashbarnwal/GSOC-2019/blob/master/doc/Accelerated_Failure_Time.pdf
 */
 
+namespace {
+
+// Allowable range for gradient and hessian. Used for regularization
+constexpr double kMinGradient = -15.0;
+constexpr double kMaxGradient = 15.0;
+constexpr double kMinHessian = 1e-16;  // Ensure that no data point gets zero hessian
+constexpr double kMaxHessian = 15.0;
+
+constexpr double kEps = 1e-12;  // A denomitor in a fraction should not be too small
+
+// Clip (limit) x to fit range [x_min, x_max].
+// If x < x_min, return x_min; if x > x_max, return x_max; if x_min <= x <= x_max, return x.
+// This function assumes x_min < x_max; behavior is undefined if this assumption does not hold.
+inline double Clip(double x, double x_min, double x_max) {
+  if (x < x_min) {
+    return x_min;
+  }
+  if (x > x_max) {
+    return x_max;
+  }
+  return x;
+}
+
+using xgboost::common::ProbabilityDistributionType;
+
+enum class CensoringType : uint8_t {
+  kUncensored, kRightCensored, kLeftCensored, kIntervalCensored
+};
+
+struct GradHessPair {
```

Comment:
Why don't you use existing gradient pair?

Reply:
Replaced with `GradientPairPrecise`.